packages/protocol/test/thirdparty/LibBytesUtils.t.sol

```diff
@@ -6,20 +6,61 @@ import "../../contracts/thirdparty/LibBytesUtils.sol";
+        assembly {
+            ptr := add(byteArray, 32)
+            mstore(ptr, v)
         }
 
-        uint256 v = uint256(0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20);
-        assertEq(LibBytesUtils.toBytes32(byteArray), bytes32(v));
+        // For i = 0 to 32, the following values are supposed to be returned.
+        //
+        //  0x0000000000000000000000000000000000000000000000000000000000000000
+        //  0x0100000000000000000000000000000000000000000000000000000000000000
+        //  0x0102000000000000000000000000000000000000000000000000000000000000
+        //  0x0102030000000000000000000000000000000000000000000000000000000000
+        //  0x0102030400000000000000000000000000000000000000000000000000000000
+        //  0x0102030405000000000000000000000000000000000000000000000000000000
+        //  0x0102030405060000000000000000000000000000000000000000000000000000
+        //  0x0102030405060700000000000000000000000000000000000000000000000000
+        //  0x0102030405060708000000000000000000000000000000000000000000000000
+        //  0x0102030405060708090000000000000000000000000000000000000000000000
+        //  0x0102030405060708090a00000000000000000000000000000000000000000000
+        //  0x0102030405060708090a0b000000000000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0000000000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d00000000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e000000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f0000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1000000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011000000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011120000000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121300000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314000000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314150000000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314151600000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314151617000000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314151617180000000000000000
+        //  0x0102030405060708090a0b0c0d0e0f1011121314151617181900000000000000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a000000000000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b0000000000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c00000000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d000000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e0000
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f00
+        //  0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20
```

Comment:
This test - imo - is incorrect.

For example, you have this variable:
`0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20`

And you want to say, that bytes1 (0x01) is the data you want to interpret (let's say this is the 'nodeId').
`0x01  -> data you wanna convert to with this to bytes32()
02030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20 -> consecutive memory area contains this 'rubbish' data
`

Padding means, (solidity is little endian!) it should look like this:
`0x0000000000000000000000000000000000000000000000000000000000000001` and not the one which is in the test.

Simple (`bytes32(input)`) will do that job as is written here:
https://docs.soliditylang.org/en/v0.8.24/types.html#conversions-between-elementary-types

<img width="782" alt="kép" src="https://github.com/taikoxyz/taiko-mono/assets/51912515/0f613466-ad59-4e4e-bee7-853ae69c4f5c">

Reply:
I think the test is right! bytesN types are padded on the right and types like uint are padded on the left. This makes some sense if you see the bytesN as just fixed size byte arrays, and so zero bytes are added to the end of the array. The original implementation also makes that clear, because the bytes were simply mloaded and so padding zeros (if not overwritten) are on the right in the `bytes` data.

But yes, `bytes32(input)` does have this exact behavior and so also passes the test! So that is also my preferred solution.